pkg/operator/status_test.go

```diff
@@ -25,3 +27,55 @@ func TestPrintOperandVersions(t *testing.T) {
 		t.Errorf("Expected: %s, got: %s", expectedOutput, got)
 	}
 }
+
+func TestOperatorStatusProgressing(t *testing.T) {
+	type tCase struct {
+		oV       []osconfigv1.OperandVersion
+		expected osconfigv1.ConditionStatus
+	}
+	tCases := []tCase{
+		{
+			oV: []osconfigv1.OperandVersion{
+				{
+					Name:    "operator",
+					Version: "1.0",
+				},
+			},
+			expected: osconfigv1.ConditionFalse,
+		},
+		{
+			oV: []osconfigv1.OperandVersion{
+				{
+					Name:    "operator",
+					Version: "2.0",
+				},
+			},
+			expected: osconfigv1.ConditionTrue,
+		},
+	}
+
+	optr := Operator{
+		operandVersions: []osconfigv1.OperandVersion{
+			{
+				Name:    "operator",
+				Version: "1.0",
+			},
+		},
+	}
+	for i, tc := range tCases {
+		co := &osconfigv1.ClusterOperator{ObjectMeta: metav1.ObjectMeta{Name: clusterOperatorName}}
+		co.Status.Versions = tc.oV
+
+		optr.osClient = fakeconfigclientset.NewSimpleClientset(co)
+		optr.statusProgressing()
+		o, _ := optr.osClient.ConfigV1().ClusterOperators().Get(clusterOperatorName, metav1.GetOptions{})
+		var condition osconfigv1.ClusterOperatorStatusCondition
+		for _, coCondition := range o.Status.Conditions {
+			if coCondition.Type == osconfigv1.OperatorProgressing {
+				condition = coCondition
+				break
+			}
+		}
+		assert.Equal(t, tc.expected, condition.Status, "test-case %v expected condition %v to be %v, but got %v", i, condition.Type, tc.expected, condition.Status)
```

Comment:
Copied this from machine-config-operator.  It prints really nice output in case of test failure.  We can remove this if we don't want to add in this dependency.

Reply:
I don't see any machine-config-operator dep :) Did you update the PR meantime?

Reply:
@ingvagabund no, I mean the 'assert' library.  They are using this library, it seems nice.  But I can just do a normal test here if we don't want to pull this in.

Reply:
(This is what machine-config-operator is using)

Reply:
I see :) I am completely fine with the new dependency.